src/test/java/uk/gov/pay/directdebit/payments/services/PaymentRequestServiceTest.java

```diff
@@ -0,0 +1,267 @@
+package uk.gov.pay.directdebit.payments.services;
+
+import com.google.common.collect.ImmutableMap;
+import org.exparity.hamcrest.date.ZonedDateTimeMatchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.directdebit.app.config.DirectDebitConfig;
+import uk.gov.pay.directdebit.app.config.LinksConfig;
+import uk.gov.pay.directdebit.payments.api.PaymentRequestResponse;
+import uk.gov.pay.directdebit.payments.dao.PaymentRequestDao;
+import uk.gov.pay.directdebit.payments.dao.PaymentRequestEventDao;
+import uk.gov.pay.directdebit.payments.dao.TokenDao;
+import uk.gov.pay.directdebit.payments.dao.TransactionDao;
+import uk.gov.pay.directdebit.payments.exception.ChargeNotFoundException;
+import uk.gov.pay.directdebit.payments.model.*;
+
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static javax.ws.rs.core.UriBuilder.fromUri;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.*;
+import static uk.gov.pay.directdebit.payments.model.PaymentRequestEvent.SupportedEvent;
+import static uk.gov.pay.directdebit.payments.model.PaymentRequestEvent.Type;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PaymentRequestServiceTest {
+    private static final String SERVICE_HOST = "http://my-service";
+    private static final long GATEWAY_ACCOUNT_ID = 1L;
+    private static final String AMOUNT = "100";
+    private static final String RETURN_URL = "http://return-service.com";
+    private static final String DESCRIPTION = "This is a description";
+    private static final String REFERENCE = "Pay reference";
+
+    private static final Map<String, String> CHARGE_REQUEST = new HashMap<String, String>() {{
+        put("amount", AMOUNT);
+        put("return_url", RETURN_URL);
+        put("description", DESCRIPTION);
+        put("reference", REFERENCE);
+    }};
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Mock
+    private TokenDao mockedTokenDao;
```

Comment:
Too many Mock implies that this class is doing way too much. But something we can discuss to do things differently for direct debit connector